README.md

```diff
@@ -1 +1,115 @@
-# HttpClientHints
\ No newline at end of file
+# MyCSharp.HttpClientHints
+
+Reading HTTP Client Hints with .NET
+
+## NuGet
+
+| NuGet |
+|-|
+| [![MyCSharp.HttpClientHints](https://img.shields.io/nuget/v/MyCSharp.HttpClientHints.svg?logo=nuget&label=MyCSharp.HttpClientHints)](https://www.nuget.org/packages/MyCSharp.HttpClientHints) |
+| [![MyCSharp.HttpClientHints.AspNetCore](https://img.shields.io/nuget/v/MyCSharp.HttpClientHints.AspNetCore.svg?logo=nuget&label=MyCSharp.HttpClientHints.AspNetCore)](https://www.nuget.org/packages/MyCSharp.HttpClientHints.AspNetCore) | `dotnet add package MyCSharp.HttpClientHints.AspNetCore` |
+
+
+## Client Hints
+
+HTTP client hints are intended to completely replace the user agents (see [mycsharp/HttpUserAgentParser](https://github.com/mycsharp/HttpUserAgentParser)) of the browser in the long term and pay more attention to data protection. We are currently in a phase in which client hints are generally regarded as expiremental and therefore do not represent a complete alternative.
+
+Furthermore, in contrast to the User-Agent header, not all client hints are provided automatically, but only on request after the first response from the client.
+
+## Usage
+
+The basic idea behind the implementation of `MyCSharp.HttpClientHints` is to have as little impact as possible on performance and middleware.
+The options are therefore registered solely via a simple configuration
+
+```csharp
+// Add Http Client Hint Options
+builder.Services.AddHttpClientHints(options =>
+  {
+      options.UserAgent = true;
+      options.Platform = true;
+      options.Architecture = true;
+      options.Device = true;
+      options.Mobile = true;
+      options.Lifetime = TimeSpan.FromDays(31);
+  });
+```
+
+The corresponding headers are set by middleware, which must also be registered.
+
+```csharp
+...
+app.UseHttpsRedirection();
+
+// Register Http Client Hints Middleware
+app.UseHttpClientHints();
+
+app.UseRouting();
+...
+```
+
+The client hints are not read by a separate dependency injection registration, but as an extension method to the header collection or to the HttpContext object.
```

Comment:
My goal was to keep the context as untouched as possible and not to enrich anything. What would be the benefit of a HTTP Feature here?

Reply:
For me it feels like the right place to a feature of the HttpContext.
The way via extension method is OK, but feels more imperative.

So the middleware adds the feature, it can be read anywhere in the following handlers.

But we can leave this open for the future.

Reply:
https://github.com/mycsharp/HttpClientHints/issues/2